sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,2050 @@
+//            // And then convert to a string as UTF-8 bytes
+//            manifest_store = store_as_bytes.map(|d| d.into_boxed_slice());
+//        }
+//
+//        // Restore the state of the reader
+//        c.pop();
+//
+//        // Return our record
+//        Ok(C2PA {
+//            majorVersion: internal_record.majorVersion,
+//            minorVersion: internal_record.minorVersion,
+//            active_manifest_uri: active_manifest_uri,
+//            manifestStore: manifest_store,
+//        })
+//    }
+//}
+}
+
+impl Default for TableC2PA {
+    fn default() -> Self {
+        Self {
+            major_version: 0,
+            minor_version: 1,
+            active_manifest_uri: Default::default(),
+            manifest_store: Default::default(),
+        }
+    }
+}
+
+/// 'head' font table. For now, there is no need for a 'raw' variant, since only
+/// byte-swapping is needed.
+#[derive(Debug)]
+#[repr(C, packed)]       // As defined by Open Font Format / OpenType (though we don't as yet directly support exotics like FIXED).
+#[allow(non_snake_case)] // As named by Open Font Format / OpenType.
+struct TableHead {
+    majorVersion: u16,
+    minorVersion: u16,
+    fontRevision: u32,
+    checksumAdjustment: u32,
+    magicNumber: u32,
+    flags: u16,
+    unitsPerEm: u16,
+    created: i64,
+    modified: i64,
+    xMin: i16,
+    yMin: i16,
+    xMax: i16,
+    yMax: i16,
+    macStyle: u16,
+    lowestRecPPEM: u16,
+    fontDirectionHint: i16,
+    indexToLocFormat: i16,
+    glyphDataFormat: i16,
+}
+
+impl TableHead {
+    /// TBD - Construct TableC2PA from a source_stream and T
+    pub fn _read<T: Read + Seek + ?Sized>(_source_stream: &mut T) -> core::result::Result<Font, Error> {
+        Err(Error::FontLoadError)?
+    }
+}
+
+/// Generic font table with unknown contents.
+#[derive(Debug)]
+struct TableUnspecified {
+    data: Vec<u8>,
+}
+
+impl TableUnspecified {
+    /// TBD - Construct TableC2PA from a source_stream and T
+    pub fn _read<T: Read + Seek + ?Sized>(_source_stream: &mut T) -> core::result::Result<Font, Error> {
+        Err(Error::FontLoadError)?
+    }
+    fn write<TDest: Write + ?Sized>(&mut self, destination: &mut TDest) -> Result<()> {
+        Ok(destination.write_all(&self.data[..])?)
+    }
+}
+
+/// Possible tables
+#[derive(Debug)]
+enum Table {
+    /// 'C2PA' table
+    C2PA(TableC2PA),
+    /// 'head' table
+    //Head(TableHead),
+    /// any other table
+    Unspecified(TableUnspecified),
+    /// WOFFHeader - not really a table
+    WoffHeader(WoffHeader),
+}
+
+/// Font abstraction sufficient for SFNT (TrueType/OpenType), WOFF (1 or 2) and
+/// EOT fonts. Potentially composable to support TrueType/OpenType Collections.
+struct Font {
+    /// Magic number for this font's container format
+    magic: Magic, //
+    /// All the Tables in this font, keyed by TableTag
+    tables: HashMap<TableTag, Table>,
```

Comment:
@brogdonm points out that if this were a `BTreeMap`, then it would stay sorted, and thereby automagically fix-up any improperly-ordered SFNT or WOFF1 containers. _(But note that in WOFF2, the table directory entries are **not** sorted by tag, but instead **define** the physical table order.)_

Reply:
👀 wow... I guess I lean back towards don't sort, read/write as given

Reply:
whoops, too late. But, can be changed.